File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
class Solution {
    public int solution(String s) {
      int answer = 0;
        for(int i=0;i<s.length();i++){
            String first=s.substring(0,1);
            String last=s.substring(1);
            s=last+first;
            String chk=new String(s);
            int result=1;

            while (chk.contains("()")||chk.contains("[]")||chk.contains("{}")){
                chk = chk.replace("()", "");
                chk = chk.replace("[]", "");
                chk = chk.replace("{}", "");
            }
            if(chk.length()>0) result=0;
            answer+=result;
        }
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    
    for _ in range(len(s)) :
        stack = []
        
        #괄호 하나씩 검사
        for ch in s :
            if ch in "([{":
                stack.append(ch)
            
            #닫는 괄호면
            else :
                if not stack :
                    break
                
                if ch == ")" and stack[-1] == "(" :
                    stack.pop()
                elif ch == "]" and stack[-1] == "[" :
                    stack.pop()
                elif ch == "}" and stack[-1] == "{" :
                    stack.pop()
                    
                else :
                    break
                    
        else : 
            if not stack :
                answer += 1
        s = s[1:] + s[0]
            
    return answer